thirdparty/runtime_environments.go

```diff
@@ -308,3 +313,110 @@ func (c *RuntimeEnvironmentsClient) getHistory(ctx context.Context, opts DistroH
 	}
 	return amiHistory, nil
 }
+
+// ImageEventEntry represents a change to the image.
+type ImageEventEntry struct {
+	Name   string
+	After  string
+	Before string
+	Type   string
+	Action string
+}
+
+// ImageEvent contains information about changes to an image when the AMI changes.
+type ImageEvent struct {
+	Entries   []ImageEventEntry
+	Timestamp time.Time
+	AMIBefore string
+	AMIAfter  string
+}
+
+// EventHistoryOptions represents the filtering arguments for getEvents. Image and Limit are required argument.
+type EventHistoryOptions struct {
+	Image string
+	Page  int
+	Limit int
+}
+
+// stringToTime converts a string representing time to type time.Time.
+func stringToTime(timeInitial string) (time.Time, error) {
+	timestamp, err := strconv.ParseInt(timeInitial, 10, 64)
+	if err != nil {
+		return time.Time{}, errors.Wrapf(err, "converting string '%s' to time", timeInitial)
+	}
+	return time.Unix(timestamp, 0), nil
+}
+
+// buildImageEventEntry make an ImageEventEntry given an ImageDiffChange.
+func buildImageEventEntry(diff ImageDiffChange) (*ImageEventEntry, error) {
+	action := ""
+	if diff.Added != "" && diff.Removed != "" {
+		action = UpdatedImageEntryAction
+	} else if diff.Added != "" {
+		action = AddedImageEntryAction
+	} else if diff.Removed != "" {
+		action = DeletedImageEntryAction
+	} else {
+		return nil, errors.New("neither added nor removed")
+	}
+	entry := ImageEventEntry{
+		Name:   diff.Name,
+		After:  diff.Added,
+		Before: diff.Removed,
+		Type:   diff.Type,
+		Action: action,
+	}
+	return &entry, nil
+}
+
+// getEvents returns information about the changes between AMIs that occurred on the image.
+func (c *RuntimeEnvironmentsClient) getEvents(ctx context.Context, opts EventHistoryOptions) ([]ImageEvent, error) {
+	if opts.Limit == 0 {
+		return nil, errors.New("no limit provided")
+	}
+	optsHistory := DistroHistoryFilterOptions{
+		Distro: opts.Image,
+		Page:   opts.Page,
+		// Add 1 to ensure that the number of ImageEvents returned matches the limit.
+		Limit: opts.Limit + 1,
+	}
+	imageHistory, err := c.getHistory(ctx, optsHistory)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting image history")
+	}
+	result := []ImageEvent{}
+	// Loop through the imageHistory which are in order from most recent to last to populate the
+	// changes between the images. We set the current index i as the AfterAMI and base the timestamp
+	// from the current index i.
+	for i := 0; i < len(imageHistory)-1; i++ {
+		amiBefore := imageHistory[i+1].AMI
+		optsImageDiffs := ImageDiffOptions{
+			BeforeAMI: amiBefore,
+			AfterAMI:  imageHistory[i].AMI,
```

Comment:
oops I notice that below we use `AMIBefore` vs in the return we use `BeforeAMI`. Could we consolidate on one? (either seems fine)